docs/XElement/sarahEdits/GettingStarted.md

```diff
@@ -0,0 +1,166 @@
+Let's create our own interactive `<Counter>` component. Note that this time, we will need to import multiple XElement components, one to represent each HTML element we wish to create.
+
+Our counter requires two buttons, as well as a counter display. We also need to create a parent `<Counter>` container element. 
+
+We can define all of these elements from XElement at once:
+
+```astro
+---
+import XElement from 'astro-`xelement`'
+const {..., button:Button, span:Display, Counter} = XElement
+---
+```
+
+Here we are defining three different types HTML elements to create four distinct HTML elements. (Remember, we're making two buttons. So they are distinct elements in the document, but they are the same type of element!) 
+
+Two types of these *Named Elements* are familiar HTML Elements: `<button>` and `<span>`. The parent component is our `<Counter>` and is not named. It produces an HTML `<counter>` element which is a `DocumentFragment`, a special type of HTML element. (reference link)
+
+Let's see how we can make our XElement `<Counter>` component **do** some stuff: Add, subtract, display the current count. This is what the code looks like:
+
+```astro
+---
+import XElement from 'astro-`xelement`'
+const {..., button:Button, span:Display, Counter} = XElement
+---
+<Counter
+    @do={(element,store)=>{
+        store.count = 0 
+    }}
+    >
+    
+    <Button 
+        @click={(event,store)=>{ 
+        display.textContent =  ++store.count
+    }}>
+        +
+    </Button>
+    
+     <Display id="display">0</Display>
+    
+    <Button 
+        @click={(event,store)=>{
+        display.textContent = --store.count
+    }}>
+        -
+    </Button>
+</Counter>
+```
+### `@` decorators
+To apply JavaScript to our component, we use one of the many `@` decorators recognized by `XElement` to specify what type of action we wish it to perform.
+
+`@do` is a common instruction to "do" something. And in our buttons, we will use `@click` to describe an `onClick` action.
+
+### `store`
+
+XElement also provides a `store`: a special non-persistent data object that is available to all `XElement` components. This lets you *store* your data and allows it to be used elsewhere on the page. We are keeping track of our counter's `count` in this store object.
+
+### `id`
+
+Giving an `id` to an XElment, as we have done with `<Display>`, allows us to target that element from elsewhere in the component tree. In this example, we can update the counter's displayed count by referencing, then changing the `<Display>` element's `textContent`.
+
+### Summary 
+In this example, what we are asking our `<Counter>` parent element to **do**  is initialise the `store` with a `count` of `0`. 
+
+And, we are telling the buttons that when they receive a `click` event, they should increment or decrement the `store.count` and update the `display.textContent`.
+
+This `<Counter>` example renders the following HTML to the page:
+```html
+<counter>
+    <span id="display">0</span>
```

Comment:
This is missing the second button 🙂

Reply:
D'oh! I was experimenting moving them around to maybe put the buttons on either side of the display, and one of them obviously never got pasted back in.